### PR TITLE
Optimize symmetry checking

### PR DESCRIPTION
**Pull request by antoine-levitt**

### CODE REVIEW

File: src/symmetry.jl [hunk 1]
```diff
@@ -48,14 +48,16 @@ function symmetry_operations(lattice, atoms, positions, magnetic_moments=[];
...
+_test_approx(x::AbstractArray{T}, y::AbstractArray{T}) where T = isapprox(x, y; atol=sqrt(eps(T)))
 """
 Filter out the symmetry operations that don't respect the symmetries of the discrete BZ grid
 """
 function symmetries_preserving_kgrid(symmetries, kcoords)
     kcoords_normalized = normalize_kpoint_coordinate.(kcoords)
-    T = eltype(kcoords[1])
-    atol = T <: Rational ? 0 : sqrt(eps(T))
-    is_approx_in(x, X) = any(y -> isapprox(x, y; atol), X)
+    is_approx_in(x, X) = any(y -> _test_approx(x, y), X)
```
[the remark this answers is not in the document]
Comment (antoine-levitt) [reply src/symmetry.jl:60]: Yeah but all these indirections are semantically meaningful. If we inline it, we'll get a all(any( that will be super hard to parse

**Comment by mfherbst** [comment]
@antoine-levitt I think this is a bit clearer.

**Comment by antoine-levitt** [comment]
Sure why not
